go/fn/go.mod

```diff
@@ -7,7 +7,8 @@ require (
 	github.com/go-errors/errors v1.0.1
 	github.com/google/go-cmp v0.5.7
 	github.com/stretchr/testify v1.7.1
-	// We must not include any core k8s modules (e.g. k8s.io/apimachinery) in
+	k8s.io/apimachinery v0.24.0
```

Comment:
Ah, didn't know the k8s.io/api before. Yes, this is exactly what we are looking for. 
I think some other parts of the SDK like the where can be simplified as well
cc @natasha41575